nuvlaedge/agent/nuvla/client_wrapper.py

```diff
@@ -306,22 +307,44 @@ def telemetry(self, new_status: dict, attributes_to_delete: list[str]) -> dict:
         return response.data
 
     def telemetry_patch(self, telemetry_jsonpatch: list, attributes_to_delete: list[str]) -> dict:
-        """ Sends telemetry metrics to the nuvlaedge-status resource using edit(put) operation
+        """Sends telemetry metrics to the nuvlaedge-status resource.
 
         Args:
             telemetry_jsonpatch: telemetry data in JSON Patch format
             attributes_to_delete: attributes no longer present in the metrics
 
-        Returns: a dict with the data of the response of the server including jobs queued for this NuvlaEdge
-
+        Returns: a dict with the data of the response of the server including
+                 jobs queued for this NuvlaEdge
         """
-        logger.debug(f"Sending telemetry patch data to Nuvla: \n {telemetry_jsonpatch}")
-        response: CimiResource = self.nuvlaedge_client.edit_patch(self.nuvlaedge_status_uuid,
-                                                                  data=telemetry_jsonpatch,
-                                                                  select=attributes_to_delete)
-        logger.debug(f"Response received from telemetry patch report: {response.data}")
+        self._log_debug_telemetry_jsonpatch(telemetry_jsonpatch,
+                                            attributes_to_delete)
+
+        response: CimiResource = self.nuvlaedge_client.edit_patch(
+            self.nuvlaedge_status_uuid,
+            data=telemetry_jsonpatch,
+            select=attributes_to_delete)
+        logger.debug("Response received from telemetry patch report: %s",
+                     response.data)
         return response.data
 
+    @staticmethod
+    def _log_debug_telemetry_jsonpatch(telemetry_jsonpatch: list,
+                                       attributes_to_delete: list[str]):
+        logger.debug("Sending telemetry patch data to Nuvla: \n %s",
+                     telemetry_jsonpatch)
+        logger.debug("Attributes no longer present in the metrics: \n %s",
+                     attributes_to_delete)
+
+        if logger.level == logging.DEBUG and len(telemetry_jsonpatch) > 0 and \
+                'op' in telemetry_jsonpatch[0] and 'path' in telemetry_jsonpatch[0]:
+            ops_paths = [(x['op'], x['path'])
+                         for x in sorted(telemetry_jsonpatch,
+                                         key=lambda x: (x['op'], x['path']))]
```

Comment:
Just a duck-typing - checking if what was given contains JSON patch operation dicts, with the healthy opportunistic assumption being that if the first element is JSON patch operation then others are as well. By definition (https://datatracker.ietf.org/doc/html/rfc6902/#section-4), 'op' and 'path' must be in the JSON patch dict.

Reply:
Ok. So why not replacing this condition by a `try...except` so that if, for whatever reason, the `telemetry_jsonpatch` is malformed, having the debug enabled will not crash the agent ?
```python
if logger.level == logging.DEBUG
    try:
        ops_paths = [(x['op'], x['path'])
                     for x in sorted(telemetry_jsonpatch,
                                     key=lambda x: (x['op'], x['path']))]
        logger.debug('Telemetry patch data ops and paths:\n%s',
                     pformat(ops_paths))
        logger.debug("Telemetry patch data size: %s",
                     len(bytes(json.dumps(telemetry_jsonpatch), 'utf-8')))
    except:
        logger.exception('Error when logging debug info on telemetry patch data')
```

Reply:
BTW this bloc is not unit-tested

Reply:
> BTW this bloc is not unit-tested

Yes, I saw this in Sonarcloud. I will add tests.

Reply:
> Ok. So why not replacing this condition by a `try...except` so that if, for whatever reason, the `telemetry_jsonpatch` is malformed, having the debug enabled will not crash the agent ?
> 
> ```python
> if logger.level == logging.DEBUG
>     try:
>         ops_paths = [(x['op'], x['path'])
>                      for x in sorted(telemetry_jsonpatch,
>                                      key=lambda x: (x['op'], x['path']))]
>         logger.debug('Telemetry patch data ops and paths:\n%s',
>                      pformat(ops_paths))
>         logger.debug("Telemetry patch data size: %s",
>                      len(bytes(json.dumps(telemetry_jsonpatch), 'utf-8')))
>     except:
>         logger.exception('Error when logging debug info on telemetry patch data')
> ```

Yep - works.

Reply:
@schaubl Please check. 
I added the unit test. I also reduced the cognitive complexity of `list_raw_resources()` (Sonar complained).